appinventor/components/src/com/google/appinventor/components/runtime/util/RuntimeErrorAlert.java

```diff
@@ -19,22 +20,41 @@
  * @author halabelson@google.com (Hal Abelson)
  */
 public final class RuntimeErrorAlert {
+
+  private static final String TAG = "RuntimeErrorAlert";
+
+
   public static void alert(final Object context,
-      final String message, final String title,final String buttonText) {
-    Log.i("RuntimeErrorAlert", "in alert");
-    AlertDialog alertDialog = new AlertDialog.Builder((Context) context).create();
-    alertDialog.setTitle(title);
-    alertDialog.setMessage(message);
-    alertDialog.setButton(buttonText, new DialogInterface.OnClickListener() {
-      public void onClick(DialogInterface dialog, int which) {
-        ((Activity) context).finish();
-      }});
+                           final String message,
+                           final String title,
+                           final String buttonText) {
+    alert(context, /* only alert dialog */ false, message, title, buttonText);
+  }
+
+  public static void alert(final Object context,
+                           boolean toast,
+                           String message,
+                           String title,
+                           String buttonText) {
+    Log.e(TAG, "alert(" + message + ", " + title + ", " + buttonText);
     if (message == null) {
-      // Avoid passing null to Log.e, which would cause a NullPointerException.
-      Log.e(RuntimeErrorAlert.class.getName(), "No error message available");
+      // do not pass null, or it will result in a crash
+      message = title + " <No error message>";
+    }
+
+    if (toast) {
+      Toast.makeText((Context) context, message, Toast.LENGTH_SHORT).show();
```

Comment:
Since this is an error situation, maybe we should prefer `LENGTH_LONG` so there is more time to read the error message.

Reply:
Done